lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -32,7 +34,8 @@
  * The interface between user code and submitted task.
  */
 @Unstable
-public final class VortexFuture<TOutput> implements Future<TOutput> {
+public final class VortexFuture<TOutput extends Serializable>
+    implements Future<TOutput>, TaskletStateDelegate<TOutput> {
```

Comment:
Sorry, actually, at the moment it seems OK, as long as it adheres to [this](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html), though I made a mistake earlier in implementing cancellation that `get` should throw  `CancellationException` if the `Future` has already been canceled. I've opened [REEF-1076](https://issues.apache.org/jira/browse/REEF-1076) to track.